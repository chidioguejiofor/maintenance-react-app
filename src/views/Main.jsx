import React from 'react';
import { Switch } from 'react-router-dom';
import routes from '../routes';

const Main = () => (
  <main>
    <Switch>
      {
        routes.map((routeObj, index) => (
          <routeObj.type
            key={index}
            exact={routeObj.exact}
            path={routeObj.routePath}
            component={routeObj.component}
          />
        ))
      }
    </Switch>
  </main>
);

export default Main;
