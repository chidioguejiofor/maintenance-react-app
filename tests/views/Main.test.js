/* eslint no-undef:off */
import React from 'react';
import { shallow } from 'enzyme';

import Main from '../../src/views/Main';

describe('Testing LandingBody Component', () => {
  it('should match existing snapshot', () => {
    expect(shallow(<Main/>)).toMatchSnapshot();
  });
});
