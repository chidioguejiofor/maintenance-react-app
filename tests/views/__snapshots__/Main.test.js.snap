// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LandingBody Component should match existing snapshot 1`] = `
<main>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="0"
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      key="1"
      path="/signup"
    />
    <Route
      component={[Function]}
      exact={true}
      key="2"
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      key="3"
      path="/admin/login"
    />
    <Route
      component={[Function]}
      exact={true}
      key="4"
      path="/reset-password"
    />
    <ProtectedRoute
      component={[Function]}
      exact={true}
      key="5"
      path="/dashboard"
    />
    <AdminRoute
      component={[Function]}
      exact={true}
      key="6"
      path="/dashboard/admin"
    />
  </Switch>
</main>
`;
